File: node-bench/loop_bench.js
const N = 100_000_000;

function main() {
    const start = Date.now();
    let sum = 0n; // BigInt, para não perder precisão
    for (let i = 0; i < N; i++) {
        sum += BigInt(i);
    }
    const end = Date.now();
    console.log("result: " + sum.toString());
    console.log("elapsed time: " + (end - start) + " ms");
}

main();

File: python-bench/loop_bench.py
import time

N = 1_000_000_000  # 100 milhões

def main():
    start = time.time()
    s = 0
    for i in range(N):
        s += i
    end = time.time()
    elapsed_ms = int((end - start) * 1000)
    print(f"result: {s}")
    print(f"elapsed time: {elapsed_ms} ms")

if __name__ == "__main__":
    main()
